src/test/resources/schema/walk-schema-default.json

```diff
@@ -10,6 +10,20 @@
           "description": "the test data does not have this attribute, so default should be applied if the ApplyDefaultsStrategy requires it",
           "type": "string",
           "default": "hello"
+        },
+        "stringValue_missing_with_default_null": {
+          "description": "the test data does not have this attribute, so default as null should be applied if the ApplyDefaultsStrategy requires it",
+          "type": "string",
+          "default": null
+        },
+        "stringValue_provided_as_null_with_default_null": {
+          "description": "the test data has this attribute as null, so default as null should be applied if the ApplyDefaultsStrategy requires it",
+          "type": "string",
+          "default": null
+        },
+        "stringValue_missing_with_no_default": {
+          "description": "the test data does not have this attribute, no default value should be applied",
+          "type": "string"
         }
```

Comment:
Do you need lines 19-27? It seems lines 14-18 is good enough.

Reply:
@SiemelNaran I removed one unnecessary schema case. But other schema case I have left from 19-21 is covering code ranging from 136 to 138 in PropertiesValidator.java file. I think this case makes sense.